File: grafos/python/provincias.py
def dfs(node, isConnected, visit):
    visit[node] = True
    for i in range(len(isConnected)):
        if isConnected[node][i] == 1 and not visit[i]:
            dfs(i, isConnected, visit)

def findCircleNum(isConnected):
    n = len(isConnected)
    numberOfComponents = 0
    visit = [False] * n

    for i in range(n):
        if not visit[i]:
            numberOfComponents += 1
            dfs(i, isConnected, visit)

    return numberOfComponents

# Exemplo 1
isConnected1 = [
    [1, 1, 0],
    [1, 1, 0],
    [0, 0, 1]
]
print(findCircleNum(isConnected1))

# Exemplo 2
isConnected2 = [
    [1, 0, 0],
    [0, 1, 0],
    [0, 0, 1]
]
print(findCircleNum(isConnected2))

File: grafos/javascript/provincias.js
var dfs = function(node, isConnected, visit) {
    visit[node] = true;
    for (let i = 0; i < isConnected.length; i++) {
        if (isConnected[node][i] === 1 && !visit[i]) {
            dfs(i, isConnected, visit);
        }
    }
}

/**
 * @param {number[][]} isConnected
 * @return {number}
 */
var findCircleNum = function(isConnected) {
    const n = isConnected.length;
    let numberOfComponents = 0;
    const visit = new Array(n).fill(false);

    for (let i = 0; i < n; i++) {
        if (!visit[i]) {
            numberOfComponents++;
            dfs(i, isConnected, visit);
        }
    }

    return numberOfComponents;
};

// Exemplo 1
let isConnected = [
    [1, 1, 0],
    [1, 1, 0],
    [0, 0, 1]
]
console.log(findCircleNum(isConnected))

// Exemplo 2
isConnected = [
    [1, 0, 0],
    [0, 1, 0],
    [0, 0, 1]
]
console.log(findCircleNum(isConnected))

